GRIDAPPSD/micro-apps
# More precommit hooks implementation

Adding tool configuration for the yapf  pre-commit hook and mypy(strict type checker) as well as performing an initial run of the hooks on all files in the repository.

How to test this specific PR from command line:

- install poetry on your system from [here](https://python-poetry.org/docs/).
- pull or merge from the develop branch of my [repository](https://github.com/afisher1/micro-apps.git).
- perform poetry update in the root repo folder.
- start the python environment by typing poetry shell.
- type pre-commit install
- type pre-commit run --all-files

expected output should look like this.

check yaml...........................................(no files to check)Skipped
fix double quoted strings............................(no files to check)Skipped
check json...........................................(no files to check)Skipped
check toml...............................................................Passed
check xml............................................(no files to check)Skipped
forbid new submodules................................(no files to check)Skipped
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
check for merge conflicts................................................Passed
don't commit to branch...................................................Passed
yapf.................................................(no files to check)Skipped
isort................................................(no files to check)Skipped

I have all of this running under my WSL 2 Ubuntu 22.04 with a min python version of 3.10 which is what is in GridAPPS-D develop container currently.

Coincidentally following these steps will initialize your dev setup for the repository :)
